pkg/tracegen/config.go

```diff
@@ -18,20 +18,138 @@
 package tracegen
 
 import (
+	"errors"
+	"fmt"
+	"math"
+	"os"
+
 	"go.elastic.co/apm/v2"
 )
 
+type ConfigOption func(Config) Config
```

Comment:
Since this function will modify the config, I think it'd be better to accept a pointer to Config and not return anything.

```suggestion
type ConfigOption func(*Config)
```